staker/block_validator.go

```diff
@@ -776,6 +796,12 @@ func (v *BlockValidator) AssumeValid(globalState validator.GoGlobalState) error
 	v.lastBlockValidatedMutex.Lock()
 	defer v.lastBlockValidatedMutex.Unlock()
 
+	v.blockMutex.Lock()
+	defer v.blockMutex.Unlock()
+
+	v.reorgMutex.Lock()
+	defer v.reorgMutex.Unlock()
```

Comment:
We lock these two mutexes in reverse order in sendRecords, which can lead to a deadlock. I think we need to document the order in which all 3 mutexes are locked (if multiple are locked at the same time). Alternatively, we should check if we can just replace them all with a single mutex.